src/pp-annotator/form/AnnotationForm.tsx

```diff
@@ -54,77 +63,187 @@ export default class AnnotationForm extends React.Component<
+      [AnnotationPriorities.ALERT]: 'priority-alert',
+    };
+    return priorityToClass[this.state.priority || AnnotationPriorities.NORMAL];
+  }
+
+  // A modal displayed when user tries to save the form with comment field empty
+  public renderNoCommentModal() {
+      this.noCommentModal = (
+          <Modal
+              open={this.state.noCommentModalOpen}
+              size="mini"
+          >
+            <Modal.Content>
+              Czy na pewno chcesz dodać przypis bez treści?
+            </Modal.Content>
+            {/* Action buttons style from semantic-ui, probably temporary */}
+            <Modal.Actions>
+              <button
+                  className="ui button negative"
+                  onClick={() => this.setState({noCommentModalOpen: false})}
+              >
+                Anuluj
+              </button>
+              <button
+                  className="ui button"
+                  onClick={this.executeSave}
+              >
+                Zapisz
+              </button>
+            </Modal.Actions>
+          </Modal>
+      );
+      return this.noCommentModal;
+  }
+
   public render() {
       const {
           priority,
           comment,
           referenceLink,
-          referenceLinkTitle
+          referenceLinkError,
+          referenceLinkTitle,
+          referenceLinkTitleError,
       } = this.state;
 
     return (
-      <div className="pp-widget">
-        <div className="pp-editor-head-bar">
-            <div className={"pp-editor-priority" + (priority=='NORMAL' ? ' selected' : '')}>
-                przypis
+        <div className="pp-widget">
+          <div className="pp-editor-head-bar">
+            <label className="priority-header"> Co dodajesz? </label>
+            <Popup
+                on="click"
+                hideOnScroll
+                trigger={<div className="priority-help"> <i className="help circle icon"></i> </div>}
+                flowing
+                hoverable
+            >
+              {/*TODO just an instruction stub*/}
+              <Grid centered divided columns={3}>
+                <Grid.Column textAlign='center'>
+                  <Header as='h4'>Niebieski przypis</Header>
+                </Grid.Column>
+                <Grid.Column textAlign='center'>
+                  <Header as='h4'>Żółty przypis</Header>
+                </Grid.Column>
+                <Grid.Column textAlign='center'>
+                  <Header as='h4'>Pomarańczowy przypis</Header>
+                </Grid.Column>
+              </Grid>
+            </Popup>
+            <br/>
+            {/*KG todo could probably be neater if done with sth like PriorityButton component*/}
+            <div className="priority-normal">
+              <button
+                  className={"pp-editor-priority" + (priority == AnnotationPriorities.NORMAL ? ' selected' : '')}
+                  onClick={() => this.setState({priority: AnnotationPriorities.NORMAL})}
```

Comment:
onClick callback is created every time this component is rerendered 🙂

Reply:
I used arrow functions for readability. Can we do it better while keeping these advantages?

Reply:
I'll rework it when doing `PriorityButton` component